### PR TITLE
IR-120: Implementing metrics for ImageStreams

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Created a prometheus Gauge to monitor the number of ImageStreamTags present in the cluster. Sums them up into two distinct groups:

- ImageStreams imported inside 'openshift' namespaces
- ImageStreams imported outside 'openshift' namespaces
- ImageStreams pushed inside 'openshift' namespaces
- ImageStreams pushed outside 'openshift' namespaces

This is the metric, as reported by prometheus:

```
image_registry_image_stream_tags_total{location="openshift",source="imported"} 368                                       
image_registry_image_stream_tags_total{location="openshift",source="pushed"} 0                                         
image_registry_image_stream_tags_total{location="other",source="imported"} 0                                         
image_registry_image_stream_tags_total{location="other",source="pushed"} 0
```

This PR also creates a prometheus metric to export the storage in use:

```
sh-4.4$ curl --insecure -s https://localhost:60000/metrics |grep storage_type
# HELP image_registry_storage_type Holds the storage in use for the image registry
# TYPE image_registry_storage_type gauge
image_registry_storage_type{storage="GCS"} 1
sh-4.4$
```

### CODE REVIEW

File: pkg/operator/metrics.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		klog.Errorf("unable to list image streams: %s", err)
+		return
+	}
+
+	var res = map[string]float64{
+		"imported_openshift": 0,
+		"pushed_openshift":   0,
+		"imported_other":     0,
+		"pushed_other":       0,
```
Comment (dmage) [inline pkg/operator/metrics.go:48]: You use it as 4 variables and never use it as a map. It's better to declare 4 variables.

Comment (ricardomaraschini) [reply pkg/operator/metrics.go]: done.

**Comment by xiuwang** [comment]
/label qe-approved
The metrics could be updated.

sh-4.4$ curl --insecure -s https://localhost:60000/metrics
# HELP image_registry_image_stream_tags_total Number of image stream tags. Source is either 'imported' or 'pushed'. 'location' label shows if the tag lives in one of the 'openshift' namespaces or 'other'
# TYPE image_registry_image_stream_tags_total gauge
image_registry_image_stream_tags_total{location="openshift",source="imported"} 370
image_registry_image_stream_tags_total{location="openshift",source="pushed"} 1
image_registry_image_stream_tags_total{location="other",source="imported"} 0
image_registry_image_stream_tags_total{location="other",source="pushed"} 1
image_registry_operator_storage_reconfigured_total 7
# HELP image_registry_storage_type Holds the storage in use for the image registry
# TYPE image_registry_storage_type gauge
image_registry_storage_type{storage="EmptyDir"} 1
image_registry_storage_type{storage="PVC"} 1
image_registry_storage_type{storage="S3"} 1

**Pull request merged**